plugins/module_utils/foreman_helper.py

```diff
@@ -1437,12 +1439,15 @@ def _flatten_entity(entity, foreman_spec):
     if entity is None:
         entity = {}
     for key, value in entity.items():
-        if key in foreman_spec and foreman_spec[key].get('ensure', True) and value is not None:
+        if key in foreman_spec and foreman_spec[key].get('ensure', True) and (value is not None or foreman_spec[key].get('type', 'str') == 'entity'):
```

Comment:
What other cases do we have, where a value can be `None`?
I think we filter all the `None`s we get from ansible...

Reply:
True, but we also (ab)use `_flatten_entity` to mangle what the API responds, and that's full of `None`.

Reply:
We could (ab-)use `False` here. Or create our own `NoEntity` constant.

Reply:
Not sure I follow, how could we end up with `False` here? The arg spec says it's a string, so it will (or `None`).

I thought about `NoEntity` and ended up thinking it's too unobvious to the user (and then differs to the list behaviour, where unset is `set to []`)

Reply:
I thought in `lookup_entity` if `""` we set it to `NoEntity`, which we define as a new class maybe, (or `False` or `""`).
Then it will not trigger the `None` test here and still work with the `if value` below.
I think the `None` behaving different if this is about entities is quite confusing.

Reply:
The user will not see `NoEntity` but still specifying `""`.

```
class NoEntity:
    def __bool__(self):
        return False
```
Should do the trick.

Reply:
Ah, okay, yeah that might work. Lemme try it.